src/cmd/create.go

```diff
@@ -0,0 +1,490 @@
+		logrus.Info("Found /etc/profile.d/toolbox.sh")
+
+		toolboxProfileBind = []string{"--volume", "/etc/profile.d/toolbox.sh:/etc/profile.d/toolbox.sh:ro"}
+		createArgs = append(createArgs, toolboxProfileBind...)
+	} else if utils.PathExists("/usr/share/profile.d/toolbox.sh") {
+		logrus.Info("Found /usr/share/profile.d/toolbox.sh")
+
+		toolboxProfileBind = []string{"--volume", "/usr/share/profile.d/toolbox.sh:/etc/profile.d/toolbox.sh:ro"}
+	} else {
+		logrus.Info("File 'toolbox.sh' does not exist in any known location")
+	}
+
+	if utils.PathExists("/media") {
+		logrus.Info("Checking if /media is a symbolic link to /run/media")
+
+		mediaPath, err := filepath.EvalSymlinks("/media")
+		if err != nil {
+			logrus.Error(err)
+		}
+
+		if mediaPath == "run/media" {
+			logrus.Info("/media is a symbolic link to /run/media")
+			command = append(command, "--media-link")
+		} else {
+			mediaBind := []string{"--volume", "/media:/media:rslave"}
+			createArgs = append(createArgs, mediaBind...)
+		}
+	}
+
+	logrus.Info("Checking if /mnt is a symbolic link to /var/mnt")
+
+	mntPath, err := filepath.EvalSymlinks("/mnt")
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	if mntPath == "var/mnt" {
+		logrus.Info("/mnt is a symbolic link to /var/mnt")
+		command = append(command, "--mnt-link")
+	} else {
+		mntBind := []string{"--volume", "/mnt:/mnt:rslave"}
+		createArgs = append(createArgs, mntBind...)
+	}
+
+	if utils.PathExists("/run/media") {
+		runMediaBind := []string{"--volume", "/run/media:/run/media:rslave"}
+		createArgs = append(createArgs, runMediaBind...)
+	}
+
+	logrus.Info("Checking if /usr is mounted read-only or read-write")
+	usrMountPoint, err := utils.GetMountPoint("/usr")
+	if err != nil {
+		logrus.Error(err)
+		logrus.Fatal("Failed to get the mount-point of /usr")
+	}
+	logrus.Infof("Mount-point of /usr is %s", usrMountPoint)
+	usrMountSourceFlags, err = utils.GetMountOptions(usrMountPoint)
+	if err != nil {
+		logrus.Error(err)
+		logrus.Fatalf("Failed to get the mount options of %s", usrMountPoint)
+	}
+	logrus.Infof("Mount flags of /usr on the host are %s", usrMountSourceFlags)
+
+	if !strings.Contains(usrMountSourceFlags, "ro") {
+		usrMountDestinationFlags = "rw"
+	}
+
+	// Inside of a toolbox we want to be able to access dbus for using flatpak-spawn and for users, who work with dbus.
+	logrus.Info("Preparing dbus system bus address")
+	dbusSystemBusAddress = strings.Split(viper.GetString("DBUS_SYSTEM_BUS_ADDRESS"), "=")[1]
+	dbusSystemBusAddress, err = filepath.EvalSymlinks(dbusSystemBusAddress)
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	dbusSystemBusAddressBind := []string{"--volume", fmt.Sprintf("%s:%s", dbusSystemBusAddress, dbusSystemBusAddress)}
+	createArgs = append(createArgs, dbusSystemBusAddressBind...)
+
+	logrus.Info("Preparing sssd-kcm socket")
+	args = []string{"show", "--value", "--property", "Listen", "sssd-kcm.socket"}
+	output, err := utils.SystemctlOutput(args...)
```

Comment:
Now that we are using a *proper* programming language, we should use systemd's D-Bus API to find out the location of the KCM socket, instead of parsing `systemctl`. D-Bus APIs are meant to be consumed programmatically, as opposed to command line utilities which are often targetted towards humans and their outputs can subtly change.

Podman uses `github.com/coreos/go-systemd`, so we could try that.